packages/jsii-pacmak/lib/targets/java.ts

```diff
@@ -1459,6 +1461,13 @@ class JavaGenerator extends Generator {
           this.code.openBlock(signature);
           let statement = '';
 
+          // do not emit these checks for primitive-only unions,
+          // because Java does not allow the check these in this case (eg if a String instanceof Number)
+          if (type.includes('java.lang.Object')) {
+            this.emitUnionParameterValdation([
+              { name: 'value', type: prop.type },
+            ]);
```

Comment:
we discussed this offline, but adding the result here; setters have overloads (one for each possible type in the union), so we have
```
setProperty(string) { }
setProperty(number) {}
```

instead of 

```setProperty(Object) {}```